File: packages/plugin-tee/src/vendors/phala.ts
import type { TeeVendor } from "./types";
import { phalaRemoteAttestationProvider as remoteAttestationProvider } from "../providers/remoteAttestationProvider";
import { phalaDeriveKeyProvider as deriveKeyProvider } from "../providers/deriveKeyProvider";
import { phalaRemoteAttestationAction as remoteAttestationAction } from "../actions/remoteAttestationAction";
import { TeeVendors } from "@elizaos/core";

/**
 * A class representing a vendor for Phala TEE.
 * * @implements { TeeVendor }
 * @type {TeeVendors.PHALA}
 *//**
 * Get the actions for the PhalaVendor.
 * * @returns { Array } An array of actions.
 *//**
 * Get the providers for the PhalaVendor.
 * * @returns { Array } An array of providers.
 *//**
 * Get the name of the PhalaVendor.
 * * @returns { string } The name of the vendor.
 *//**
 * Get the description of the PhalaVendor.
 * * @returns { string } The description of the vendor.
 */
export class PhalaVendor implements TeeVendor {
	type = TeeVendors.PHALA;

	/**
	 * Returns an array of actions.
	 *
	 * @returns {Array} An array containing the remote attestation action.
	 */
	getActions() {
		return [remoteAttestationAction];
	}
	/**
	 * Retrieve the list of providers.
	 *
	 * @returns {Array<Function>} An array containing two provider functions: deriveKeyProvider and remoteAttestationProvider.
	 */
	getProviders() {
		return [deriveKeyProvider, remoteAttestationProvider];
	}

	/**
	 * Returns the name of the plugin.
	 * @returns {string} The name of the plugin.
	 */
	getName() {
		return "phala-tee-plugin";
	}

	/**
	 * Get the description of the function
	 * @returns {string} The description of the function
	 */
	getDescription() {
		return "Phala TEE Cloud to Host Eliza Agents";
	}
}

File: plugins/plugin-tee/python/elizaos_plugin_tee/vendors/phala.py
from __future__ import annotations

from elizaos_plugin_tee.actions.remote_attestation import REMOTE_ATTESTATION_ACTION
from elizaos_plugin_tee.providers import (
    get_derived_keys,
    get_remote_attestation,
)
from elizaos_plugin_tee.vendors.types import TeeVendorInterface, TeeVendorNames


class PhalaVendor(TeeVendorInterface):
    @property
    def type(self) -> str:
        return TeeVendorNames.PHALA

    def get_actions(self) -> list[dict[str, object]]:
        return [REMOTE_ATTESTATION_ACTION]

    def get_providers(self) -> list[object]:
        return [get_derived_keys, get_remote_attestation]

    def get_name(self) -> str:
        return "phala-tee-plugin"

    def get_description(self) -> str:
        return "Phala Network TEE for secure agent execution"
